### PR TITLE
fix: properly handle token extraction

### PR DESCRIPTION
**Pull request by guillaumelamirand**
**Description**

Properly handle token extraction

- no Authorization header will not return any security token
- no Bearer security schema value will not return a security token (Authorization: Basic 123456)
- Empty Bearer will return an invalid security token
- token without client_id will now return an invalid Security Token

**Additional context**

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `3.0.0-properly-handle-empty-token-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-oauth2/3.0.0-properly-handle-empty-token-SNAPSHOT/gravitee-policy-oauth2-3.0.0-properly-handle-empty-token-SNAPSHOT.zip)

### CODE REVIEW

File: src/test/java/io/gravitee/policy/oauth2/utils/TokenExtractorTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        Assertions.assertNull(token);
+    }
+
+    @Test
+    void should_extract_with_authorization_header_and_empty_bearer() {
+        HttpHeaders headers = HttpHeaders.create().set("Authorization", io.gravitee.policy.jwt.utils.TokenExtractor.BEARER);
+        when(request.headers()).thenReturn(headers);
+
+        String token = io.gravitee.policy.jwt.utils.TokenExtractor.extract(request);
+        Assertions.assertNotNull(token);
```
Comment (jhaeyaert) [inline src/test/java/io/gravitee/policy/oauth2/utils/TokenExtractorTest.java:70]: assertEquals("") ?

File: src/test/java/io/gravitee/policy/oauth2/utils/TokenExtractorTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        when(request.headers()).thenReturn(headers);
+
+        String token = io.gravitee.policy.jwt.utils.TokenExtractor.extract(request);
+
+        Assertions.assertNotNull(token);
+        Assertions.assertEquals(jwt, token);
+    }
+
+    @Test
+    void should_extract_from_query_parameter() {
```
Comment (jhaeyaert) [inline src/test/java/io/gravitee/policy/oauth2/utils/TokenExtractorTest.java:100]: Should we add a test with an empty query param value?

**Review by jhaeyaert — APPROVED** [review]
Few comments but looks ok.

File: src/test/java/io/gravitee/policy/oauth2/DummyOAuth2Resource.java [hunk 3]
```diff
@@ -46,6 +46,8 @@ public void introspect(String accessToken, Handler<OAuth2Response> responseHandl
             response = new OAuth2Response(true, "{this _is _invalid json");
         } else if (TOKEN_FAIL.equals(accessToken)) {
             response = new OAuth2Response(false, null);
+        } else {
+            response = new OAuth2Response(false, null);
```
[the remark this answers is not in the document]
Comment (guillaumelamirand) [reply src/test/java/io/gravitee/policy/oauth2/DummyOAuth2Resource.java:50]: hum no, where do you see one? it is to handle "empty token"
